pkg/alertmanager/operator.go

```diff
@@ -475,6 +726,281 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+func (c *Operator) provisionAlertmanagerConfiguration(ctx context.Context, am *monitoringv1.Alertmanager, store *assetStore) error {
+
+	secretName := defaultConfigSecretName(am.Name)
+	if am.Spec.ConfigSecret != "" {
+		secretName = am.Spec.ConfigSecret
+	}
+
+	secret, err := c.kclient.CoreV1().Secrets(am.Namespace).Get(ctx, secretName, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrap(err, "get base configuration secret")
+	}
+
+	var rawBaseConfig []byte
+	if secret != nil && secret.Data["alertmanager.yaml"] != nil {
+		rawBaseConfig = secret.Data["alertmanager.yaml"]
+	} else {
+		level.Warn(c.logger).Log("msg", "either base config secret not found, or alertmanager.yaml field not specified",
+			"secret", secretName, "alertmanager", am.Name, "namespace", am.Namespace)
+
+		fallbackConfig := "route:\n  receiver: 'null'\n" +
+			"receivers:\n- name: 'null'"
+		rawBaseConfig = []byte(fallbackConfig)
+	}
+
+	baseConfig, err := loadCfg(string(rawBaseConfig))
+	if err != nil {
+		return errors.Wrap(err, "base config from Secret could not be parsed")
+	}
+
+	// If the base route is missing from the secret, then alertmanager will
+	// fail to load the config (and we'll fail to add subroutes)
+	foundRoute := baseConfig.Route != nil
```

Comment:
Alertmanager config loader should prevent that.

Reply:
Good point, if it was nil we'd never get to this point. I've updated it now.